src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs

```diff
@@ -244,14 +244,23 @@ protected virtual void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanColl
                     // is there any effect on the view?
                     if (viewLines.IntersectsBufferSpan(changedSpan))
                     {
-                        AdornmentLayer.RemoveAdornmentsByVisualSpan(changedSpan);
+                        RemoveAdornment(viewLines, changedSpan);
                     }
                 }
             }
 
             AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(changedSpanCollection);
         }
 
+        /// <summary>
+        /// Removes the adornment based on the changed span. 
+        /// Needs special logic for InlineDiagnostics since the span added is different.
+        /// </summary>
+        protected virtual void RemoveAdornment(IWpfTextViewLineCollection viewLines, SnapshotSpan changedSpan)
```

Comment:
this shoudl be abstract.  we want all subclasses to have to think about what the right behavior is.